### PR TITLE
Remove json_name field when json_name is specified as default

### PR DESCRIPTION
**Pull request by hhkkxxx133**
There is a a [known bug](https://github.com/protocolbuffers/protobuf/issues/5587) with protoc library that `json_name` will always be populated no matter if it is specified. This will be inconsistent with the Apicurio generated version as well as generated by protoc POJO compiler`protoc -I=. --java_out=out/ file.proto`, where the attribute `json_name` is missing if it is not specified.  

In this case, we propose the following solution to make sure the parsed schema will always be the same no matter what generation method is used:
1. if a default `json_name` is specified in the schema, it will be ignored. 
2. if a non-default `json_name` is specified in the schema, it will be kept as usual.

In addition, we would like to know if there would be a recent release on the library.

### CODE REVIEW

File: utils/protobuf-schema-utilities/src/main/java/io/apicurio/registry/utils/protobuf/schema/FileDescriptorUtils.java [hunk 1]
```diff
@@ -463,7 +463,8 @@ private static DescriptorProto messageElementToDescriptorProto(
                     allNestedTypes.put(field.getLocation(), mapMessage.build());
                 }
 
-                String jsonName = field.getDeclaredJsonName();
+                String jsonName = getDefaultJsonName(field.getName()).equals(field.getDeclaredJsonName())
+                        ? null : field.getDeclaredJsonName();
```
Comment (blacktooth) [inline utils/protobuf-schema-utilities/src/main/java/io/apicurio/registry/utils/protobuf/schema/FileDescriptorUtils.java:467]: Is this applicable for map fields?

Comment (hhkkxxx133) [reply utils/protobuf-schema-utilities/src/main/java/io/apicurio/registry/utils/protobuf/schema/FileDescriptorUtils.java:467]: Yes it is applicable for both Map itself as well as Map key/value fields. Map itself is tested. Map key/value fields are auto generated and can also be taken care of by the logic.

**Review by blacktooth — APPROVED** [review]

**Comment by hhkkxxx133** [comment]
@carlesarnal Thanks for you help! May I ask when the next release of the library is planned? Or is it possible to release protobuf-schema-utilities module so that we can have the fix integrated to our project please? Much appreciated!

**Comment by carlesarnal** [comment]
cc @EricWittmann wdyt?

**Comment by EricWittmann** [comment]
@hhkkxxx133 Sorry for my silence on this one.  It slipped through my email filters.  :(  But as I'm guessing you know, we've done a new release of registry so you should be all set now.

**Pull request merged**